src/__tests__/integration/data/thethar/thethar-contract-wrc.js

```diff
@@ -0,0 +1,403 @@
+(() => {
+    // src/thetAR/actions/common.ts
+    var isAddress = (addr) => /[a-z0-9_-]{43}/i.test(addr);
+    var hashCheck = async (validHashs, contractTxId) => {
+      const tx = await SmartWeave.unsafeClient.transactions.get(contractTxId);
+      let SrcTxId;
+      tx.get("tags").forEach((tag) => {
+        let key = tag.get("name", { decode: true, string: true });
+        if (key === "Contract-Src") {
+          SrcTxId = tag.get("value", { decode: true, string: true });
+        }
+      });
+      if (!SrcTxId || !isAddress(SrcTxId)) {
+        throw new ContractError("Cannot find valid srcTxId in contract Tx content!");
+      }
+      const srcTx = await SmartWeave.unsafeClient.transactions.getData(SrcTxId, { decode: true, string: true });
+      if (srcTx.length < 1e4 && validHashs.includes(calcHash(srcTx))) {
+        return true;
+      }
+      return false;
+    };
+    var calcHash = (string) => {
+      var hash = 0, i, chr;
+      if (string.length === 0)
+        return hash;
+      for (i = 0; i < string.length; i++) {
+        chr = string.charCodeAt(i);
+        hash = (hash << 5) - hash + chr;
+        hash |= 0;
```

Comment:
Isn't `|= 0` basically noop?

Reply:
again, it's a compiled code (probably generated by the TS compiler), so hard for me to say.

BUT...I believe the idea here is to use '0' as the 'default' value, in case the previous assignments would result in sth like NaN, null, or whatever.